src/common/utxobased/engine/makeUtxoEngine.ts

```diff
@@ -296,7 +302,10 @@ export async function makeUtxoEngine(
       )
       const utxos =
         options?.utxos ??
-        ((await processor.fetchUtxos({ utxoIds: [] })) as IUTXO[])
+        ((await processor.fetchUtxos({
+          scriptPubkey: utxoScriptPubkey,
+          utxoIds: []
+        })) as IUTXO[])
```

Comment:
The type assertion isn't trustworthy anymore now that we've parameterized `fetchUtxo` differently. Use `filterUndefined` utility function to actually remove undefined values from the return value.